src/meta/src/hummock/compaction/compaction_picker.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_pb::hummock::Level;
+
+use crate::hummock::compaction::overlap_strategy::OverlapStrategy;
+use crate::hummock::compaction::SearchResult;
+use crate::hummock::level_handler::LevelHandler;
+
+pub trait CompactionPicker {
```

Comment:
Shall we provide a `need_compaction`?

Reply:
No. `LevelSelector` will do this job. If the current LSM tree is balanced, we will get `None` when calling `pick_compaction`.

Reply:
I think `LevelSelector` is OK.

- `HummockManager::commit_epoch/HummockManager::report_compact_task` will initiate a compaction request to compaction scheduler if there is no pending compaction request for the same compaction group.

- The compaction scheduler consumes compaction request and keeps calling `pick_compaction` for the same compaction group until got `None`.